src/js/actions/mask.js

```diff
@@ -142,6 +150,98 @@ define(function (require, exports) {
             menuActions.native]
     };
 
+    /**
+     * Create an circular mask the size of the currently selected layer
+     *
+     * @return {Promise}
+     */
+    var applyRectangle = function () {
+        var appStore = this.flux.store("application"),
+            toolStore = this.flux.store("tool"),
+            currentDocument = appStore.getCurrentDocument();
+
+        if (!currentDocument) {
+            return Promise.resolve();
+        }
+
+        var currentLayers = currentDocument.layers.selected;
+
+        if (currentLayers.isEmpty()) {
+            return Promise.resolve();
+        }
+
+        var currentLayer = currentLayers.first(),
+            bounds = boundsLib.bounds(currentLayer.bounds),
```

Comment:
If this is going to be applicable to not just single layers, but layer groups as well, you probably want to use `currentDocument.layers.childBounds(currentLayer)` here, otherwise it's ok the way it is.